skglm/tests/test_estimators.py

```diff
@@ -97,27 +97,38 @@ def test_check_estimator(estimator_name):
 # Test if skglm solver returns the coefficients
 @pytest.mark.parametrize("estimator_name", dict_estimators_ours.keys())
 @pytest.mark.parametrize('X', [X, X_sparse])
-def test_estimator(estimator_name, X):
+@pytest.mark.parametrize('fit_intercept', [True, False])
+def test_estimator(estimator_name, X, fit_intercept):
     if estimator_name == "GeneralizedLinearEstimator":
         pytest.skip()
     estimator_sk = dict_estimators_sk[estimator_name]
     estimator_ours = dict_estimators_ours[estimator_name]
+    # TODO This seems a bit unusal, maybe to discuss
+    if fit_intercept:
+        estimator_sk.fit_intercpet = True
+        estimator_ours.fit_intercpet = True
     estimator_sk.fit(X, y)
     estimator_ours.fit(X, y)
     coef_sk = estimator_sk.coef_
     coef_ours = estimator_ours.coef_
     # assert that something was fitted:
+    if fit_intercept:
+        # add test intercept from fit_intercept
+        estimator_sk.fit_intercpet = False
```

Comment:
typo intercpet

this design should be avoided, it is dangerous on many levels (global modification of variable, sensitive to typos such as here)

Reply:
agreed. I don't understand why we need this hack btw. I tried running the tests while commenting these lines and it fails. Why?